### PR TITLE
Add CI/CD recommendations

### PR DESCRIPTION
**Pull request by olivomarco**
Consider caching when using CI/CD agents and split large code repositories

### CODE REVIEW

**Comment by jborsecnik** [comment]
@Zimmergren, 

Can you review the proposed changes? 

IMPORTANT: When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by Zimmergren** [comment]
Thank you for the submission @olivomarco. I'll review and format the content and pass it on to the WAF Sustainability v-team for approval.

File: well-architected/sustainability/sustainability-testing.md [hunk 1]
```diff
@@ -46,6 +46,28 @@ _Green Software Foundation alignment: [Hardware efficiency](sustainability-desig
 - Only scale out when necessary, and when not testing, scale in. Ultimately this ensures there's no idle compute resources in test environments.
 - Consider optimized platform services like containers over testing in a VM, utilizing the platform to reduce maintenance.
 
+## Consider caching when using CI/CD agents
```
Comment (Zimmergren) [inline well-architected/sustainability/sustainability-testing.md:49]: Requesting changes to this section. Here's a formatted snippet you can use for this text block. 

```
### Consider caching when using CI/CD agents

Using caching mechanisms during CI/CD can reduce compute time and, thus, carbon emissions.

_Green Software Foundation alignment: [Energy Efficiency](sustainability-design-principles.md#energy-efficiency)_

**Recommendation:**

- Store results from steps in a cache and re-use them between different CI/CD runs when possible.
- If the CI/CD agent is self-hosted, use a cache local to the agent to further reduce data transfers and emissions.

### Split large code repositories

Splitting large repositories can help the CI/CD phases, where only the parts of the code that have changed are compiled. This reduces compute time, which ultimately lowers carbon emissions.

_Green Software Foundation alignment: [Energy Efficiency](sustainability-design-principles.md#energy-efficiency)_

**Recommendation:**

- Split large code repositories, separating main code from libraries and dependencies.
- Publish and re-use artifacts and libraries of code that are common across multiple repositories.
```

**Review by Zimmergren — CHANGES_REQUESTED** [review]

File: well-architected/sustainability/sustainability-testing.md (hunk 1 above)
Comment (olivomarco) [reply well-architected/sustainability/sustainability-testing.md]: Ok, I have committed to my patch branch

**Comment by cmcclister** [comment]
Learn Build status updates of commit _[42d83b6](https://github.com/olivomarco/well-architected/commits/42d83b6b8d2cf8970c377bbf41b71c93248ed38d)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[well-architected/sustainability/sustainability-testing.md](https://github.com/olivomarco/well-architected/blob/patch-1/well-architected/sustainability/sustainability-testing.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/915b64c6-eecb-d366-4a84-4831c5e1bfda/202210260822291744-53/BuildReport?accessString=a0be491dd9205e31827d9ad81c08301a925c2b6948b9f03fcd4a5ef26f0275b6).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/915b64c6-eecb-d366-4a84-4831c5e1bfda?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

**Comment by cmcclister** [comment]
Learn Build status updates of commit _[020073f](https://github.com/olivomarco/well-architected/commits/020073f02d90a93f1e8be22a7d9a2c18b91a4cc5)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[well-architected/sustainability/sustainability-testing.md](https://github.com/olivomarco/well-architected/blob/patch-1/well-architected/sustainability/sustainability-testing.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/915b64c6-eecb-d366-4a84-4831c5e1bfda/202210261128198035-53/BuildReport?accessString=bb1f94a9c9cee0bbc39139a41c0399a3a688746c1a92755b4a0b2b3de51a8e0f).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/915b64c6-eecb-d366-4a84-4831c5e1bfda?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

**Review by Zimmergren — APPROVED** [review]

**Comment by Zimmergren** [comment]
#sign-off

**Review by Court72 — APPROVED** [review]

**Pull request merged**